src/components/Table/GridTable.tsx

```diff
@@ -558,36 +585,53 @@ function renderVirtual<R extends Kinded>(
         return (maybeContentsDiv.firstElementChild! as HTMLElement).offsetHeight;
       }}
       itemContent={(index) => {
-        // We keep header and filter rows separate, but react-virtuoso is a flat list,
-        // so we pick the right header / first row message / actual row.
-        let i = index;
-        if (i < headerRows.length) {
-          return headerRows[i][1];
+        // Since we have two arrays of rows: `headerRows` and `filteredRow` we
+        // must determine which one to render.
+
+        // Determine if we need to render a header row
+        if (index < headerRows.length) {
+          return headerRows[index][1];
         }
-        i -= headerRows.length;
+
+        // Reset index
+        index -= headerRows.length;
+
+        // Show firstRowMessage as the first `filteredRow`
         if (firstRowMessage) {
-          if (i === 0) {
+          if (index === 0) {
             return (
               <div css={Css.add("gridColumn", `${columns.length} span`).$}>
                 <div css={{ ...style.firstRowMessageCss }}>{firstRowMessage}</div>
               </div>
             );
           }
-          i -= 1;
+
+          // Shift index -1 when there is a firstRowMessage to not skip the
+          // first `filteredRow`
+          index--;
         }
-        return filteredRows[i][1];
+
+        // Lastly render `filteredRow`
+        return filteredRows[index][1];
       }}
       totalCount={(headerRows.length || 0) + (firstRowMessage ? 1 : 0) + (filteredRows.length || 0)}
     />
   );
 }
 
 /**
- * Customizes the `List` element that react-virtuoso renders, to have our css grid logic.
+ * A table might render two of these components to represent two virtual lists.
```

Comment:
This was unclear when first trying to solve this problem. Adding this comment for future reference.